2312/2/battlers_table.rb

```diff
@@ -0,0 +1,37 @@
+require 'terminal-table'
+
+# BattlersTable prints a table of data, you provide to it
+class BattlersTable
+  def initialize(battlers, num_top)
+    @battlers = battlers
+    @counter = -1
+    @table = []
+    @table = create_table(num_top)
```

Comment:
Ideally, you should not place any logic in the initializer.
It should set initial state only.

You allowed to do this preparation in `#print` method if needed.

```ruby
def print
  create_table
  do_something_useful

  puts Terminal::Table.new rows: table
end
```